### PR TITLE
Undeploy for Crontab

### PR DESCRIPTION
**Pull request by gilleyj**
reference jira DEVOPS-66 https://imagineeasy.atlassian.net/browse/DEVOPS-66

### CODE REVIEW

File: easybib/providers/crontab.rb [hunk 1]
```diff
@@ -66,3 +66,31 @@
   new_resource.updated_by_last_action(updated)
 
 end
+
+action :delete do
+  app = new_resource.app
+
+  updated = false
```
Comment (till) [inline easybib/providers/crontab.rb]: You can remove `updated` and always return true.

File: easybib/providers/crontab.rb [hunk 2]
```diff
@@ -66,3 +66,31 @@
...
+
+  execute 'Clear old cron.d files' do
+    # rm will exit with 1 if no old cron.d files existed
+    # adding a "; true" to remove the loooong warning in chef logs everyone stumbles upon
+    command "rm /etc/cron.d/#{app}_*; true"
+    ignore_failure true
+  end
+
+  execute 'Clear old crontab' do
+    updated = true
```
Comment (till) [inline easybib/providers/crontab.rb]: This as well.

File: easybib/providers/crontab.rb [hunk 3]
```diff
@@ -66,3 +66,31 @@
...
+    user node['nginx-app']['user']
+    # crontab will exit with 130 if crontab has already been cleared
+    # adding a "; true" to remove the loooong warning in chef logs everyone stumbles upon
+    command "crontab -u #{node['nginx-app']['user']} -r; true"
```
Comment (till) [inline easybib/providers/crontab.rb]: Make an attribute for `node['nginx-app']['user']` and inject that value instead please.

File: ies/recipes/opsworks-undeploy.rb [hunk 4]
```diff
@@ -0,0 +1,3 @@
+if is_aws
```
Comment (till) [inline ies/recipes/opsworks-undeploy.rb]:
```
unless is_aws
  return
end
```

@gilleyj maybe we do a quick test-run for un-deploy on a test stack? I can't find JSON output for it anywhere.

Comment (till) [inline ies/recipes/opsworks-undeploy.rb]:
```
easybib_crontab ...

 action :delete
end
```

File: easybib/providers/crontab.rb [hunk 5]
```diff
@@ -66,3 +66,29 @@
   new_resource.updated_by_last_action(updated)
 
 end
+
+action :delete do
+  app = new_resource.app
+  crontab_user = new_resource.crontab_user
```
Comment (till) [inline easybib/providers/crontab.rb]: Maybe check on if it was `nil` somewhere. But this looks good.

File: ies/recipes/role-undeploy.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+include_recipe 'easybib::crontab'
+
+unless is_aws
+  node['deploy'].each do |application, deploy|
+    easybib_crontab application do
+      app application
```
Comment (till) [inline ies/recipes/role-undeploy.rb]: `crontab_user` missing

**Comment by gilleyj** [comment]
Recipe worked in a dupe of QA stack using satis as a app. Worked on recipe execute ies::role-undeploy... Worked with app delete, app undeploy and app deploy.
Keep in mind for the undeploy event to be realized, the layer will require the "UNDEPLOY" recipe to be realized as "ies::role-undeploy"

**Pull request merged**